Runtime/Game/Requests/LootLockerLeaderboardDetails.cs

```diff
@@ -0,0 +1,152 @@
+        public bool enable_game_api_writes { get; set; }
+        public bool overwrite_score_on_submit { get; set; }
+        public bool has_metadata { get; set; }
+        public LootLockerLeaderboardSchedule schedule { get; set; }
+
+        public LootLockerLeaderboardReward[] rewards { get; set; }
+
+    }
+
+    public class LootLockerLeaderboardRewardCurrency
+    {
+        public string created_at { get; set; }
+        public string updated_at { get; set; }
+        public string amount { get; set; }
+        public LootLockerLeaderboardRewardCurrencyDetails details { get; set; }
+        public string reward_id { get; set; }
+        public string currency_id { get; set; }
+    }
+
+    public class LootLockerLeaderboardRewardCurrencyDetails
+    {
+        public string name { get; set; }
+        public string code { get; set; }
+        public string amount { get; set; }
+        public string id { get; set; }
+    }
+
+    public class LootLockerLeaderboardRewardProgression
+    {
+        public string created_at { get; set; }
+        public string updated_at { get; set; }
+        public LootLockerLeaderboardRewardPorgressionDetails details { get; set; }
+        public int amount { get; set; }
+        public string progression_id { get; set; }
+        public string reward_id { get; set; }
+
+    }
+
+    public class LootLockerLeaderboardRewardPorgressionDetails
+    {
+        public string key { get; set; }
+        public string name { get; set; }
+        public int amount { get; set; }
+        public string id { get; set; }
+    }
+
+    public class LootLockerLeaderboardRewardProgressionReset
+    {
+        public string created_at { get; set; }
+        public string updated_at { get; set; }
+        public LootLockerLeaderboardRewardProgressionResetDetails details { get; set; }
+        public string progression_id { get; set; }
+        public string reward_id { get; set; }
+    }
+    public class LootLockerLeaderboardRewardProgressionResetDetails
+    {
+        public string key { get; set; }
+        public string name { get; set; }
+        public string id { get; set; }
+    }
+
+    public class LootLockerLeaderboardRewardAsset
+    {
+        public string created_at { get; set; }
+        public string updated_at { get; set; }
+        public LootLockerLeaderboardRewardAssetDetails details { get; set; }
+        public string asset_variation_id { get; set; }
+        public string asset_rental_option_id { get; set; }
+        public int asset_id { get; set; }
+        public string reward_id { get; set; }
+        public string asset_ulid { get; set; }
+
+    }
+
+    public class LootLockerLeaderboardRewardAssetDetails
+    {
+        public string name { get; set; }
+        public string thumbnail { get; set; }
+        public string variation_name { get; set; }
+        public string rental_option_name { get; set; }
+        public int? variation_id { get; set; }
```

Comment:
Is there an important point to having variation_id and rental_option_id both inside and outside the details class? Seems unneccessary at a first glance. And the inner is nullable ints, but the outer are strings, why the type difference for the same data?